src/main/java/com/atlassian/db/replica/api/aurora/AuroraReplicasDiscoverer.java

```diff
@@ -0,0 +1,55 @@
+package com.atlassian.db.replica.api.aurora;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.atlassian.db.replica.api.aurora.AuroraEndpoints.replicaUrl;
+
+/**
+ * It allows to pull Aurora Replicas cluster information
+ */
+public class AuroraReplicasDiscoverer {
+    private final Connection connection;
+
+    public AuroraReplicasDiscoverer(Connection connection) {
+        this.connection = connection;
+    }
+
+    /**
+     * It provides list of replica urls based on reader endpoint and server ids
+     * @param readerEndpoint - Aurora reader endpoint
+     * @return list of replicas jdbcUrls
```

Comment:
Perhaps it should return objects rather than Strings? It would be a form of fail fast. Note that JDBC URLs aren't valid `URI`s directly due to the prefix, but it's ok: 

```java
    @Test
    public void shouldParseJdbcUrl() {
        JdbcUrl parsed = JdbcUrl.parse("jdbc:postgresql://x.y.z:5432/db?x=y");

        assertThat(parsed.getUrl().getHost()).isEqualTo("x.y.z");
    }

    @Test
    public void shouldReParseJdbcUrl() {
        String raw = "jdbc:postgresql://x.y.z:5432/db?x=y";

        Throwable thrown = catchThrowable(() -> JdbcUrl.parse(JdbcUrl.parse(raw).toString()));

        assertThat(thrown).doesNotThrowAnyException();
    }

    class JdbcUrl {

        private static final String PREFIX = "jdbc:";

        private final URI url;

        private JdbcUrl(URI url) {
            this.url = url;
        }

        public URI getUrl() {
            return url;
        }

        public static JdbcUrl parse(String rawJdbcUrl) {
            if (rawJdbcUrl.startsWith(PREFIX)) {
                URI parsed = URI.create(rawJdbcUrl.replaceFirst(PREFIX, ""));
                return new JdbcUrl(parsed);
            } else {
                String parseError = String.format("Cannot parse %s due to missing prefix '%s'", rawJdbcUrl, PREFIX);
                throw new IllegalArgumentException(parseError);
            }
        }

        @Override
        public String toString() {
            return PREFIX + url.toString();
        }
    }
```

PS. we got a very similar class in the consumer of this lib.

Reply:
Now that I think about it... It's not JdbcUrl. It's `instance endpoint` according to [AWS definitions](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Overview.Endpoints.html#Aurora.Overview.Endpoints.Types) (it has nothing to do with Jdbc).

Reply:
I'm also good with using `parse` instead of `fromString` 👌 It's more Java-ish